### PR TITLE
fix(datetimepicker): resolve issue with active cell highlighting

### PR DESCRIPTION
**Pull request by bah99**
- **Enhanced rendering and performance** by updating the `@for` `track` function in **calendar-body.html**.  
- **Streamlined active cell and date selection handling** in **month-view.html** for a more efficient user experience.  
- **Added `_getActiveCell` method** in **month-view.ts** to prioritize highlighting of the selected date or today's date when no date is selected in the calendar view.

### CODE REVIEW

File: projects/extensions/datetimepicker/calendar-body.html [hunk 1]
```diff
@@ -6,39 +6,41 @@
...
+        [attr.aria-selected]="selectedValue === item.value"
+      >
+        {{ item.displayValue }}
+      </div>
+    </td>
     }
   </tr>
-}
+  }
+
```
Comment (Kounda-Dartagnan) [inline projects/extensions/datetimepicker/calendar-body.html:46]: new empty line

**Review by Kounda-Dartagnan — CHANGES_REQUESTED** [review]

File: projects/extensions/datetimepicker/month-view.html [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
   </thead>
-  <tbody mtx-calendar-body
+  <tbody
+    mtx-calendar-body
     (@slideCalendar.done)="_calendarStateDone()"
     [@slideCalendar]="_calendarState"
     [rows]="_weeks"
     [todayValue]="_todayDate!"
-    [activeCell]="_adapter.getDate(activeDate) - 1"
+    [activeCell]="_getActiveCell()"
```
Comment (nzbin) [inline projects/extensions/datetimepicker/month-view.html:15]: The active cell logic has no issue, we just should improve the a11y.
https://github.com/angular/components/blob/main/src/material/datepicker/month-view.html#L25

Comment (bah99) [reply projects/extensions/datetimepicker/month-view.html:15]: Yes, I figured it out! Material has implemented a11y very effectively.

**Comment by bah99** [comment]
I’m working to improve accessibility by implementing the same approach as the Material Date Picker and ensuring it functions in a similar way.

**Comment by bah99** [comment]
Hello @nzbin , 

I've successfully implemented a behavior similar to the Angular Material datepicker navigation. Currently, this is implemented for the month view (type='date'). 

Could you please review the implementation and provide your feedback? If it meets the requirements, I'll extend the same approach to the year and multi-year views.

Looking forward to your thoughts!

**Comment by nzbin** [comment]
> Hello @nzbin ,
> 
> I've successfully implemented a behavior similar to the Angular Material datepicker navigation. Currently, this is implemented for the month view (type='date').
> 
> Could you please review the implementation and provide your feedback? If it meets the requirements, I'll extend the same approach to the year and multi-year views.
> 
> Looking forward to your thoughts!

Do you check the Angular Material source codes? You need the CDK a11y module to implement and there're lots of work.
https://github.com/angular/components/blob/9f73fed4702f79a6461fd2cf86ecb311c9530a78/src/material/datepicker/calendar.html#L3
https://github.com/angular/components/blob/9f73fed4702f79a6461fd2cf86ecb311c9530a78/src/material/datepicker/calendar-body.scss#L286

**Comment by bah99** [comment]
> > Hello @nzbin ,
> > I've successfully implemented a behavior similar to the Angular Material datepicker navigation. Currently, this is implemented for the month view (type='date').
> > Could you please review the implementation and provide your feedback? If it meets the requirements, I'll extend the same approach to the year and multi-year views.
> > Looking forward to your thoughts!
> 
> Do you check the Angular Material source codes? You need the CDK a11y module to implement and there're lots of work. https://github.com/angular/components/blob/9f73fed4702f79a6461fd2cf86ecb311c9530a78/src/material/datepicker/calendar.html#L3 https://github.com/angular/components/blob/9f73fed4702f79a6461fd2cf86ecb311c9530a78/src/material/datepicker/calendar-body.scss#L286

Yes, sometimes. I realized that using the a11y module would require significant time and effort, so I implemented a solution that achieves the same behavior in a simpler way.

**Comment by nzbin** [comment]
@bah99 I have fixed this issue in v19.

**Pull request closed without merging**